### DISCUSSION TITLE
Pushing the output of a static build site is in conflict with gitignore

### DISCUSSION THREAD

**Discussion by rosenbergm** [dokku/dokku#7087] [General]
Hey there, people of Dokku!

I just tried to deploy a static site with the Nginx buildpack and after some very _tough_
moments, I got it working!

Now I've stumbled upon a new "issue". See, my build system (zola) produces a
`public` directory with the website contents. This directory is, logically, also in my
`.gitignore` file. I don't want my repository polluted.

However, to deploy the static content, it needs to be staged and committed, so it can
be pushed to the Dokku remote.

How do you solve this kind of a problem? Is the best solution indeed using `git add -f`?

Cheers!

**Comment by josegonzalez** [comment]
Typically you'll run that build process in the image you are building, so the build command would happen in the push (or on your remote CI process if going down that route). I see that zola is a rust app... do you mind sharing how you are building your app so we can help you modify that process to work with Dokku?

**Reply by rosenbergm** [reply]
Hey, yes, sure!

For now, my deployment is done from my local 
machine (no CI).

1. Build the webpage with `zola build -o dist`
(the _dist_ is important, it's not in my `.gitignore`.
2. Now I stage the changes, commit them and push 
to my Dokku remote

My question rather is: Am I able to push my 
built data to Dokku and avoid committing them 
into my development repository?

Would this be solved by running it in a CI 
pipeline? I don't see how because the gitignore
file would still be there.

Thanks!

**Reply by rosenbergm** [reply]
Oh and also, I'm using the Nginx Buildpack for
this website I'm deploying.
